### PR TITLE
Support network load balancer

### PR DESCRIPTION
**Pull request by elviejokike**
@npalm Let's check this one next week together.
Mainly, this change will allow to create use load balancers of type "network". By default, "application" will be used as it is currently doing.

### CODE REVIEW

**Review by npalm — COMMENTED** [review]
Looks great! 
- Please can you add an example for NLB
- What is the rationale for adding an NLB?

File: alb.tf [hunk 1]
```diff
@@ -36,9 +36,33 @@ resource "aws_security_group" "security_group_alb" {
   }
 }
 
+locals {
+  enable_alb = "${var.enable_alb * (var.alb_load_balancer_type == "application" ? 1 : 0)}"
+  enable_nlb = "${var.enable_alb * (var.alb_load_balancer_type == "network"? 1 : 0)}"
+}
+
+resource "aws_alb" "nlb" {
+  // Only enable if ALB is required
```
Comment (npalm) [inline alb.tf]: Update comment, this in NLB or not?

File: variables.tf [hunk 2]
```diff
@@ -84,6 +84,11 @@ variable "alb_port" {
   default     = 443
 }
 
+variable "alb_load_balancer_type" {
+  description = "Load balancer type"
```
Comment (npalm) [inline variables.tf]: Please describe the options

**Review by npalm — CHANGES_REQUESTED** [review]

**Comment by elviejokike** [comment]
Network load balancers are useful, when using api gateways. If you want to connect AWS api gateway to resources in your VPC, network load balancer is the available alternative.

**Comment by npalm** [comment]
@elviejokike Looks good!
- Is it possible to add an example in examples?
- Can you update the changelog?

**Comment by npalm** [comment]
As discussed we should not create a VPC link for each service. We only need one per VPC. We can make the service load balancer enabled so the consumer of the module can connect the right load balancer and set up the VPC link. I will close the issue

**Pull request closed without merging**